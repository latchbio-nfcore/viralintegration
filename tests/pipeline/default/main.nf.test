nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.trace.tasks().size(),

                path("$outputDir/summary/test.vif.prelim.refined.tsv"),
                // FIXME md5sum fails on CI
                // path("$outputDir/summary/test.vif.refined.distilled.tsv"),
                // path("$outputDir/summary/test.vif.refined.tsv"),
                // path("$outputDir/summary/test.vif.refined.wRefGeneAnnots.tsv"),

                path("$outputDir/virus/novirus.VirusDetect.igvjs.bam"),
                path("$outputDir/virus/novirus.VirusDetect.sorted.igvjs.bam"),
                path("$outputDir/virus/test.VirusDetect.igvjs.bam"),
                path("$outputDir/virus/test.VirusDetect.sorted.igvjs.bam"),
                path("$outputDir/virus/test.VirusDetect.virus_read_counts_summary.tsv"),
            ).match()

            assert new File("$outputDir/summary/test.vif.refined.distilled.tsv").exists()
            assert new File("$outputDir/summary/test.vif.refined.tsv").exists()
            assert new File("$outputDir/summary/test.vif.refined.wRefGeneAnnots.tsv").exists()

            assert new File("$outputDir/summary/test.vif.genome_plot.png").exists()
            assert new File("$outputDir/summary/test.vif.html").exists()

            assert new File("$outputDir/virus/novirus.VirusDetect.init.genome_plot.png").exists()
            assert new File("$outputDir/virus/novirus.VirusDetect.virus_read_counts.png").exists()
            assert new File("$outputDir/virus/novirus.VirusDetect.virus_read_counts_log.png").exists()
            assert new File("$outputDir/virus/test.VirusDetect.igvjs.html").exists()
            assert new File("$outputDir/virus/test.VirusDetect.init.genome_plot.png").exists()
            assert new File("$outputDir/virus/test.VirusDetect.virus_read_counts.png").exists()
            assert new File("$outputDir/virus/test.VirusDetect.virus_read_counts_log.png").exists()
        }

    }

}
