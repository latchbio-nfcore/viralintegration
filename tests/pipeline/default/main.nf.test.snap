{
    "Should run without failures": {
        "content": [
            33,
            "test.vif.prelim.refined.tsv:md5,17b26b7a86d61206154ca3e2df02fc48",
            "novirus.VirusDetect.igvjs.bam:md5,1ace945eb2ce9ec48b3de1988cf26ba4",
            "novirus.VirusDetect.sorted.igvjs.bam:md5,ac314f2d8b013e136c829e166806a182",
            "test.VirusDetect.igvjs.bam:md5,8462d7f485afea5cd33eff2bb1611cd2",
            "test.VirusDetect.sorted.igvjs.bam:md5,56f176c542e02f88779737105a864150",
            "test.VirusDetect.virus_read_counts_summary.tsv:md5,c051eb1a25e4741c43274589f75b80b6"
        ],
        "timestamp": "2023-11-28T20:39:55+0000"
    }
}
